python/dgl/dataloading/pytorch/dataloader.py

```diff
@@ -325,6 +325,51 @@ def __next__(self):
         result = [_to_device(data, self.device) for data in result_]
         return result
 
+class _DistDataLoaderWrapper:
+    """
+    A wrapper for DistDataLoader, copy features from original DistGraph to blocks
+    """
+    def __init__(self, g, dataloader):
+        self.g = g
+        assert isinstance(g, DistGraph), "Input g must be a DistGraph."
+        self.dataloader = dataloader
+
+    def __iter__(self):
+        self.dataloader = iter(self.dataloader)
+        return self
+
+    def __next__(self):
+        try:
+            ret = list(next(self.dataloader))
+            blocks = []
+            for _, block in enumerate(ret[-1]):
+                # copy node and edge features
+                blocks.append(self._copy_features(block))
+            ret[-1] = blocks
+            return tuple(ret)
+        except StopIteration:
+            raise StopIteration
+
+    def _copy_features(self, block):
+        """copy DistTensor reference to block features"""
+        # copy node feature DistTensor reference
+        for ntype in block.ntypes:
+            nids = block.nodes(ntype)
+            block_nodes = block.nodes[ntype]
+            g_nodes = self.g.nodes[ntype]
+            for name in self.g._get_ndata_names(ntype):
+                ndata_name = name.get_name()
+                block_nodes.data[ndata_name] = (g_nodes.data[ndata_name], nids)
+        # copy edge feature DistTensor reference
+        for etype in block.etypes:
+            eids = block.edges[etype].data[EID]
+            block_edges = block.edges[etype]
+            g_edges = self.g.edges[etype]
+            for name in self.g._get_edata_names(etype):
+                edata_name = name.get_name()
+                block_edges.data[edata_name] = (g_edges.data[edata_name], eids)
+        return block
```

Comment:
You may find some internal utility functions like [`extract_node_subframes_for_block`](https://github.com/dmlc/dgl/blob/master/python/dgl/utils/internal.py#L793) designed for similar purposes. Ideally, you don't need to change the `__init__` function of a `Column`. The `subframe` method will save the index array for you.

Reply:
I revised `_copy_features()` but didn't use `extract_node_subframes_for_block()` directly, since `DistGraph` doesn't have [self._node_frames](https://github.com/dmlc/dgl/blob/master/python/dgl/utils/internal.py#L820). ( For `DistGraph`, it is `self._ndata_store`.) So `DistTensor` is copied to blocks first and then block is used as `DGLHeteroGraph`. I am not sure if it is a good idea to write a new function like `extract_node_subframes_for_block()` but for DistGraph:
```
node_frames = []
for i, ind_nodes in enumerate(srcnodes):
    data = {}
    ntype = dist_g.ntypes[i]
    for name in dist_g._get_ndata_names(ntype):
        data[name.get_name()] = dist_g.nodes[ntype].data[name.get_name()]
    nframe = Frame(data, ind_nodes)
    subf = nframe.subframe(ind_nodes)
    node_frames.append(subf)
```
With this, `_copy_features()` can be simplified. Which approach seems more appropriate to you? Do you have any better ideas?